app/views/layouts/_header.html.erb

```diff
@@ -0,0 +1,32 @@
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
+  <%= link_to 'Upcoming Events', events_path, class: 'btn btn-info' %>
+  <%= link_to "Past Events", events_filter_past_path, class: 'btn btn-secondary ml-2' %>
+  <div class="collapse navbar-collapse" id="navbarNav">
+    <ul class="navbar-nav">
+    <% if current_user %>
+      <li class="nav-item">
+        <%= link_to current_user.name, user_path(current_user), class: 'nav-link btn'  %>
+      </li>
+
+    <li class="nav-item">
+      <%= link_to 'Sign Out', session_path, method: :delete, class: 'nav-link' %> 
+    </li>
+  <% else %>
+    <li class="nav-item">
+      <%= link_to 'Sign Up', new_user_path, class: 'nav-link' %> 
+    </li>
+
+    <li class="nav-item">
+      <%= link_to 'Sign In', new_session_path, class: 'nav-link' %>
+    </li>
+  <% end %>
```

Comment:
- [ ] create a helper method that has the if/else logic and render the links and markup within the helper method instead of doing it straight in the view you can also use partials when you need to render several lines within a condition in your helper method